addons/cert.py

```diff
@@ -234,6 +234,24 @@ def msc30(data):
         if simpleMatch(token, "rand ( )") and isStandardFunction(token):
             reportError(token, 'style', 'Do not use the rand() function for generating pseudorandom numbers', 'MSC30-c')
 
+# STR03-C
+# Do not inadvertently truncate a string
+def str03(data):
+    for token in data.tokenlist:
+        if simpleMatch(token, 'strncpy ('): # now search if the 3rd parameter is a sizeof()
```

Comment:
I would suggest a early continue here.

    if not simpleMatch(token, 'strncpy ('):
        continue